main/src/ca/uwaterloo/flix/language/ast/ops/SimplifiedAstOps.scala

```diff
@@ -513,4 +515,291 @@ object SimplifiedAstOps {
+        val r = renameBoundVars(rest, vars0, labels0)
+        Expression.RecordExtend(label, v, r, tpe, loc)
+
+      case Expression.RecordRestrict(label, rest, tpe, loc) =>
+        val r = renameBoundVars(rest, vars0, labels0)
+        Expression.RecordRestrict(label, r, tpe, loc)
+
+      case Expression.ArrayLit(elms, tpe, loc) =>
+        val es = elms map (renameBoundVars(_, vars0, labels0))
+        Expression.ArrayLit(es, tpe, loc)
+
+      case Expression.ArrayNew(elm, len, tpe, loc) =>
+        val e = renameBoundVars(elm, vars0, labels0)
+        val ln = renameBoundVars(len, vars0, labels0)
+        Expression.ArrayNew(e, ln, tpe, loc)
+
+      case Expression.ArrayLoad(base, index, tpe, loc) =>
+        val b = renameBoundVars(base, vars0, labels0)
+        val i = renameBoundVars(index, vars0, labels0)
+        Expression.ArrayLoad(b, i, tpe, loc)
+
+      case Expression.ArrayStore(base, index, elm, tpe, loc) =>
+        val b = renameBoundVars(base, vars0, labels0)
+        val i = renameBoundVars(index, vars0, labels0)
+        val e = renameBoundVars(elm, vars0, labels0)
+        Expression.ArrayStore(b, i, e, tpe, loc)
+
+      case Expression.ArrayLength(base, tpe, loc) =>
+        val b = renameBoundVars(base, vars0, labels0)
+        Expression.ArrayLength(b, tpe, loc)
+
+      case Expression.ArraySlice(base, startIndex, endIndex, tpe, loc) =>
+        val b = renameBoundVars(base, vars0, labels0)
+        val i1 = renameBoundVars(startIndex, vars0, labels0)
+        val i2 = renameBoundVars(endIndex, vars0, labels0)
+        Expression.ArraySlice(b, i1, i2, tpe, loc)
+
+      case Expression.Ref(exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.Ref(e, tpe, loc)
+
+      case Expression.Deref(exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.Deref(e, tpe, loc)
+
+      case Expression.Assign(exp1, exp2, tpe, loc) =>
+        val e1 = renameBoundVars(exp1, vars0, labels0)
+        val e2 = renameBoundVars(exp2, vars0, labels0)
+        Expression.Assign(e1, e2, tpe, loc)
+
+      case Expression.Existential(fparam0, exp, loc) =>
+        val fparam1 = fparam0.copy(sym = Symbol.freshVarSym(fparam0.sym))
+        val vars1 = vars0 + (fparam0.sym -> fparam1.sym)
+        val e = renameBoundVars(exp, vars1, labels0)
+        Expression.Existential(fparam1, e, loc)
+
+      case Expression.Universal(fparam0, exp, loc) =>
+        val fparam1 = fparam0.copy(sym = Symbol.freshVarSym(fparam0.sym))
+        val vars1 = vars0 + (fparam0.sym -> fparam1.sym)
+        val e = renameBoundVars(exp, vars1, labels0)
+        Expression.Universal(fparam1, e, loc)
+
+      case Expression.Cast(exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.Cast(e, tpe, loc)
+
+      case Expression.TryCatch(exp, rules, tpe, loc) =>
+        // TODO: is this right or do we need to handle it like Branch?
```

Comment:
Whats the real question? :)

Reply:
I don't really know how this construct works. I'm imagining you have a single expression (exp) wrapped in a try ... catch ... catch .. where each catch corresponds to a CatchRule.

So the symbol in the CatchRule cannot occur in exp or any of the other CatchRules. Unlike a Branch where a "case-symbol" can occur across cases. Is that right?

Reply:
> So the symbol in the CatchRule cannot occur in exp or any of the other CatchRules. 
Correct.